### PR TITLE
Add support for condensing mobs and merge all portal-spawned mobs into a single type.

### PR DESCRIPTION
**Pull request by xFier**
Added support for Mob Condensing, where spawns of a mob are reduced by a factor, but items dropped by that type of mob are increased by the same factor.

Stopped spawning Wither Skeletons and Ghasts from Nether Portals to replace them with Zombified Piglin variants.

Wither Skeletons will now be Zombie Piglins wearing a Wither Skull as a hat, and Ghasts will be Zombie Piglins holding a Ghast Tear in their offhand.

There should be no changes to game balance, aside from a reduction in mob counts.

PortalSpawnModifer requires no config changes.
Mob Condenser requires a config in the following format:
MobCondenser:
  enabled: true
  mobSpawnModifiers:
    ZOMBIFIED_PIGLIN: 0.10

The mobSpawnModifiers controls the entity type, and the new spawn rate, so in the above example Zombified Piglins will spawn 10x less, but drop 10x the items.

### CODE REVIEW

File: src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/MobCondenser.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+				}
+			}
+		}
+	}
+
+	@EventHandler
+	public void onMobEggUse(PlayerInteractEvent e) {
+		if (e.getAction() == Action.RIGHT_CLICK_BLOCK && e.getItem() != null) {
+			try {
+				EntityType type = EntityType.valueOf(e.getItem().getType().toString().replace("_SPAWN_EGG", ""));
```
Comment (Protonull) [inline src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/MobCondenser.java:87]: You can do: 
```java
EntityType type = SpawnEggUtils.getEntityType(e.getItem().getType());
```

Comment (xFier) [reply src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/MobCondenser.java]: Thanks, I must've not seen that when I looked to see if there was something that does that.

**Comment by Diet-Cola** [comment]
How does this handle a donkey with a chest attached with an inventory full of diamond blocks?

**Comment by xFier** [comment]
> How does this handle a donkey with a chest attached with an inventory full of diamond blocks?

Probably how you'd expect it to. Not sure why you'd want to condense a donkey though

**Comment by Quinndolin** [comment]
> 
> 
> How does this handle a donkey with a chest attached with an inventory full of diamond blocks?

a tale as old as time

**Comment by okx-code** [comment]
I don't understand why wither skeletons need to be zombie piglins now?

**Comment by Virizion** [comment]
> I don't understand why wither skeletons need to be zombie piglins now?

mainly to be able to use piglin AI for farms

**Comment by Diet-Cola** [comment]
Just had another thought with this, wither skeles and pigmen can swap swords with players stuff right, so this could also multiple those drops

**Comment by xFier** [comment]
> Just had another thought with this, wither skeles and pigmen can swap swords with players stuff right, so this could also multiple those drops

Piglin item pickup is disabled, which should stop that.

**Comment by Ameliorate** [comment]
Instead of held items, shouldn't this use something like `PersistantDataContainer`s or NBT tags? Using PDCs or NBTs would completely prevent the possibility of dupe bugs.

**Comment by wingzero54** [comment]
Where/ how does the material whitelist get set/added? Is it an additional config entry?

**Comment by xFier** [comment]
> Where/ how does the material whitelist get set/added? Is it an additional config entry?

Yes, here’s an example config:
MobCondenser:
    enabled: true
    mobSpawnModifiers:
      ZOMBIFIED_PIGLIN: 0.10
      ZOMBIE: 0.50
    materialModificationWhitelist:
      - GHAST_TEAR
      - WITHER_SKELETON_SKULL
      - GUNPOWDER
      - BONE
      - COAL
      - ROTTEN_FLESH
      - GOLD_NUGGET
  PortalSpawnModifier:
    enabled: true
    pigManChance: 0.15
    ghastChance: 0.01
    witherSkeletonChance: 0.05
    witherHeadDropChance: 0.04

**Comment by Ameliorate** [comment]
It's kind of a moot point now that it's merged, but I think that this PR will cause issues in the future that will lead to duping. Even though we have proved that it is impossible to dupe using this feature in the present, the civ codebase will change. I think that problems will especially arise in the mid-far future when we update to a new version of minecraft. We won't look at this code in context of the future changes, and it will break.

If this code breaks, it causes dupe bugs. If I was in charge of reviewing, I would have not merged this PR. We should be very careful when writing code that generates multiplied items, and this code is not very careful. As I stated above, this code should be using something unaccessable to the players, such as `PersistantDataContainer`s or NBT tags.

**Comment by wingzero54** [comment]
> 
> 
> It's kind of a moot point now that it's merged, but I think that this PR will cause issues in the future that will lead to duping. Even though we have proved that it is impossible to dupe using this feature in the present, the civ codebase will change. I think that problems will especially arise in the mid-far future when we update to a new version of minecraft. We won't look at this code in context of the future changes, and it will break.
> 
> If this code breaks, it causes dupe bugs. If I was in charge of reviewing, I would have not merged this PR. We should be very careful when writing code that generates multiplied items, and this code is not very careful. As I stated above, this code should be using something unaccessable to the players, such as `PersistantDataContainer`s or NBT tags.

I agree and it's something to keep a close eye on. If you come up with some PR's as improvements we would consider them

**Comment by xFier** [comment]
> It's kind of a moot point now that it's merged, but I think that this PR will cause issues in the future that will lead to duping. Even though we have proved that it is impossible to dupe using this feature in the present, the civ codebase will change. I think that problems will especially arise in the mid-far future when we update to a new version of minecraft. We won't look at this code in context of the future changes, and it will break.
> 
> If this code breaks, it causes dupe bugs. If I was in charge of reviewing, I would have not merged this PR. We should be very careful when writing code that generates multiplied items, and this code is not very careful. As I stated above, this code should be using something unaccessable to the players, such as `PersistantDataContainer`s or NBT tags.

I'm going to have to agree too, I'll likely make another PR addressing the use of PDRs and any other suggestions you have until you're happy with the state of the code.

**Pull request merged**